### PR TITLE
cephadm: Fix type errors reported by pyright

### PR DESCRIPTION
**Pull request by sebastian-philipp**
pyright is picky about `except ImportError` constructs
and also checks un-annotated code by default.

Unfortunately, I was not able to properly add pyright into our CI: https://github.com/microsoft/pyright/issues/1364 

See also #38739

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -898,12 +895,6 @@ def check_ip_port(ip, port):
 # https://github.com/benediktschmitt/py-filelock/blob/master/filelock.py
 # that drops all of the compatibility (this is Unix/Linux only).
 
-try:
-    TimeoutError
-except NameError:
-    TimeoutError = OSError
-
```
Comment (sebastian-philipp) [inline src/cephadm/cephadm:905]: https://github.com/microsoft/pyright/issues/1363

**Review by jecluis — APPROVED** [review]

**Comment by sebastian-philipp** [comment]
```
cephadm: note: In function "kernel_security":
cephadm:5793: error: Missing return statement
cephadm:5811: error: Missing return statement
Found 2 errors in 1 file (checked 1 source file)
```

**Comment by jecluis** [comment]
those don't seem related to this patchset?

**Comment by sebastian-philipp** [comment]
> those don't seem related to this patchset?

mypy is picky about implicit `return None` at end of functions:

**Review by jmolmo — APPROVED** [review]
LGTM

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-01-13_11:19:08-rados:cephadm-wip-swagner3-testing-2021-01-12-1316-distro-basic-smithi/

Lots of failures, but they seem to be unrelated to this PR:
* https://tracker.ceph.com/issues/48873
* https://tracker.ceph.com/issues/48715
* https://tracker.ceph.com/issues/48142
* https://github.com/ceph/ceph/pull/38844#issuecomment-760089828

**Comment by jecluis** [comment]
@sebastian-philipp can we please merge this then? ;)

**Pull request merged**